kpatch-build/kpatch-elf.c

```diff
@@ -829,6 +829,8 @@ void kpatch_elf_teardown(struct kpatch_elf *kelf)
 	struct rela *rela, *saferela;
 
 	list_for_each_entry_safe(sec, safesec, &kelf->sections, list) {
+		if (sec->ignore)
+			continue;
```

Comment:
Why was this part needed?

Reply:
Without it...

```
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff78683dd in __isoc99_fscanf () from /lib64/libc.so.6
Missing separate debuginfos, use: dnf debuginfo-install elfutils-libelf-0.168-1.fc24.x86_64 zlib-1.2.8-10.fc24.x86_64
(gdb) bt
#0  0x00007ffff78683dd in __isoc99_fscanf () from /lib64/libc.so.6
#1  0x000000000040d2e6 in symvers_read (table=0x733d60, path=0x7fffffffe73d "/root/.kpatch/obj/Module.symvers")
    at lookup.c:277
#2  0x000000000040d553 in lookup_open (obj_path=0x7fffffffe70c "/root/.kpatch/tmp/vmlinux", 
    symvers_path=0x7fffffffe73d "/root/.kpatch/obj/Module.symvers", hint=0x7ffff7fc0589 "array.c", locals=0x733ca0)
    at lookup.c:322
#3  0x0000000000408e53 in main (argc=7, argv=0x7fffffffe3f8) at create-diff-object.c:2738
```

Give me some time to backtrace to figure out why it's only needed after the patch.  I imagine there is something special about the discard sections vs. the previous use of the `sec->ignore` flag.

Reply:
Without it:
```
*** Error in `/root/kpatch.joe/kpatch-build/create-diff-object': free(): invalid next size (fast): 0x00000000006b5e90 ***
...
Program received signal SIGABRT, Aborted.
0x00007ffff7833738 in raise () from /lib64/libc.so.6
Missing separate debuginfos, use: dnf debuginfo-install libgcc-6.3.1-1.fc24.x86_64
(gdb) bt
#0  0x00007ffff7833738 in raise () from /lib64/libc.so.6
#1  0x00007ffff783534a in abort () from /lib64/libc.so.6
#2  0x00007ffff78746f0 in __libc_message () from /lib64/libc.so.6
#3  0x00007ffff787ce7a in _int_free () from /lib64/libc.so.6
#4  0x00007ffff78803dc in free () from /lib64/libc.so.6
#5  0x000000000040c659 in kpatch_elf_teardown (kelf=0x6a9630) at kpatch-elf.c:918
#6  0x0000000000408ffd in main (argc=7, argv=0x7fffffffe408) at create-diff-object.c:2794
```
Give me some time to backtrack to figure out why this patch's use of `sec->ignore` is different than before.